spec/lucky_web/text_helpers_spec.cr

```diff
@@ -0,0 +1,294 @@
+require "../../spec_helper"
+
+class TextHelpers::Index < LuckyWeb::Action
+  action { render_text "foo" }
+end
+
+class TextHelpers::Create < LuckyWeb::Action
+  action { render_text "foo" }
+end
+
+private class TestPage
+  include LuckyWeb::HTMLPage
+
+  def render
+  end
+
+  def test_truncate
+    truncate "Hello World", length: 8 do
+      link "Continue", "#"
+    end
+  end
+
+  def test_highlight
+    highlight "This is a beautiful morning, but also a beautiful day", "beautiful" do |word|
```

Comment:
Same question with this one as well

Reply:
This block is slightly different than how truncate acts but similar in purpose. In the case of highlight, the block is passed to gsub as the substitution. The block is definitely a matter of preference since you can accomplish the replace another way (as with gsub).

Beauty is in the eye of the beholder but perhaps this contrived example illustrates the utility of the block form a bit more.

```crystal
text = "This is a beautiful morning, but also a beautiful day"
highlight text, "beautiful" do |word|
  span word, "data-highlight-word": word, "data-color": "yellow", "data-español": "bello"
end
```
vs.
```crystal
text = "This is a beautiful morning, but also a beautiful day"
highlight text, "beautiful", highlighter: span(word, "data-highlight-word": word, "data-color": "yellow", "data-español": "bello").to_s
```

Admittedly contrived but the latter is a bit of a word jumble on one line to me.